File: Wander.Server/Scripts/Game/ServerPlayer.ts
﻿class ServerPlayer extends Player {
    newPosition: Phaser.Point;

    constructor(state: GameState, game: Phaser.Game, pseudo: string, position: Phaser.Point, sex: number) {
        super(state, game, pseudo, position, sex);
        this.newPosition = new Phaser.Point(this.position.x, this.position.y);
    }

    update() {
        
        super.update();

        this.texture.body.x = Lerp(this.newPosition.x, this.texture.body.x, 1.35);
        this.texture.body.y = Lerp(this.newPosition.y, this.texture.body.y, 1.35);

        if (this.texture.position.x === this.newPosition.x && this.texture.position.y === this.newPosition.y) {
            if (this.direction != EDirection.Idle) {
                this.direction = EDirection.Idle;
                super.updateAnimationFrame();
            }
            
        }       
    }

    updateInfos(newPosition: Phaser.Point, newDirection: EDirection) {
        var distX = Math.abs(this.texture.position.x - newPosition.x);
        var distY = Math.abs(this.texture.position.y - newPosition.y);
        if (distX >= 10 || distY >= 10) {
            this.texture.position.x = newPosition.x;
            this.texture.position.y = newPosition.y;
            this.direction = newDirection;
        }
        this.newPosition = newPosition;
        if (this.direction != newDirection) {
            this.direction = newDirection;
            super.updateAnimationFrame();
        }

    }
}

File: Wander.Android.App/core/src/com/wander/game/models/ServerPlayer.java
package com.wander.game.models;

import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Vector2;
import com.wander.game.MainGame;

/**
 * Created by valentin on 08/12/2015.
 */
public class ServerPlayer extends Player {

    Vector2 newPosition;
    public ServerPlayer(MainGame game, String pseudo, Vector2 position, Sprite sprite) {
        super(game, pseudo, position, sprite);
        this.newPosition = position;
    }

    @Override
    public void update(float delta)
    {
        super.update(delta);
        this.position = this.position.lerp(new Vector2(this.newPosition.x, this.newPosition.y),1.35f);
    }

    @Override
    public void render(SpriteBatch batch)
    {
        super.render(batch);
    }

    public void updateInfos(Vector2 newPosition, String newDirection)
    {
        this.newPosition = newPosition;
    }
}
